### ISSUE TITLE
is it possible to link this to other LLM?

### ISSUE THREAD

**Issue by ramixpe** [poloclub/transformer-explainer#22]
is it possible to link this to other LLM?

**Comment by jan-fischer-hawhamburg** [comment]
yes at long its same architecture (gpt2) 

see my fork , it uses gerpt2 https://huggingface.co/benjamin/gerpt2

**Comment by aereeeee** [comment]
Hi @ramixpe. We are currently using a GPT-2 ONNX model via the [onnxruntime-web API](https://onnxruntime.ai/docs/tutorials/web/). Our model outputs logits and attention data for input tensors, and we use the logits to generate prediction outputs through a softmax sampling process.

If your ONNX model also returns logits and attention data for input tensors, it should work with our app. However, you'll need to modify the code in [page.svelte](https://github.com/poloclub/transformer-explainer/blob/main/src/routes/%2Bpage.svelte#L39), where the model is imported, and in [data.ts](https://github.com/poloclub/transformer-explainer/blob/main/src/utils/data.ts#L102-L121), where the inference results are processed to match your model.

Also Thank you for your answer @jan-fischer-hawhamburg!

**Comment by aereeeee** [comment]
I’d like to add a clarification. My previous response only considered ensuring that the app would work without breaking when using your model. Currently, the app cannot visualize the inner workings of your model. This visualization specifically tailored on the inner workings of GPT-2 small (12 layers, 12 attention heads, 768-dimensional embeddings). Please keep this in mind!
